lib/modules/manager/terraform/lockfile/hash.ts

```diff
@@ -23,44 +23,88 @@ export class TerraformProviderHash {
 
   static hashCacheTTL = 10080; // in minutes == 1 week
 
-  private static async hashFiles(files: string[]): Promise<string> {
+  private static async hashElementList(
+    basePath: string,
+    fileSystemEntries: string[],
+  ): Promise<string> {
     const rootHash = crypto.createHash('sha256');
 
-    for (const file of files) {
+    for (const entryPath of fileSystemEntries) {
+      const absolutePath = upath.resolve(basePath, entryPath);
+      if (!(await fs.cachePathIsFile(absolutePath))) {
+        continue;
+      }
+
       // build for every file a line looking like "aaaaaaaaaaaaaaa  file.txt\n"
-      const hash = crypto.createHash('sha256');
 
-      // a sha256sum displayed as lowercase hex string to root hash
-      const fileBuffer = await fs.readCacheFile(file);
+      // get hash of specific file
+      const hash = crypto.createHash('sha256');
+      const fileBuffer = await fs.readCacheFile(absolutePath);
       hash.update(fileBuffer);
-      rootHash.update(hash.digest('hex'));
 
-      // add double space, the filename and a new line char
-      rootHash.update('  ');
-      const fileName = file.replace(regEx(/^.*[/]/), '');
-      rootHash.update(fileName);
-      rootHash.update('\n');
+      const line = util.format('%s  %s\n', hash.digest('hex'), entryPath);
```

Comment:
@rarkins WDYT? we never used this function before. We always used simple string templates.

Reply:
I prefer to stick to string templates, but don't feel strongly about it